src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -347,11 +348,23 @@ class ProtobufPlugin implements Plugin<Project> {
         variant.registerJavaGeneratingTask(generateProtoTask.get(), generateProtoTask.get().outputSourceDirectories)
 
         project.plugins.withId("org.jetbrains.kotlin.android") {
-          project.afterEvaluate {
-            String compileKotlinTaskName = Utils.getKotlinAndroidCompileTaskName(project, variant.name)
-            project.tasks.named(compileKotlinTaskName, KotlinCompile) { KotlinCompile task ->
-              task.dependsOn(generateProtoTask)
-              task.source(generateProtoTask.get().outputSourceDirectories)
+          // Checking if Kotlin plugin is a recent one - 1.7.20+
+          if (it.respondsTo("getPluginVersion")) {
+            def kotlinExtension = project.extensions.getByType(KotlinAndroidProjectExtension.class)
+            kotlinExtension.target.compilations.named(variant.name) {
+              defaultSourceSet.kotlin.srcDir(
+                project.objects.fileCollection().from(
+                  generateProtoTask.map { it.outputSourceDirectories }
```

Comment:
Use `get()` like we did previously. It doesn't hurt anything as we are required to call get() to pass to registerJavaGeneratingTask. And there's https://github.com/gradle/gradle/issues/21179 . I'm surprised it is still an issue, honestly. Maybe one day the world will be a nicer place.

Reply:
Done :heavy_check_mark: